src/libproc_macro/lib.rs

```diff
@@ -200,6 +201,72 @@ impl Span {
     pub fn call_site() -> Span {
         ::__internal::with_sess(|(_, mark)| Span(mark.expn_info().unwrap().call_site))
     }
+
+    /// The original source file into which this span points.
+    #[unstable(feature = "proc_macro", issue = "38356")]
+    pub fn file(&self) -> SourceFile {
```

Comment:
Should this return `Option<SourceFile>` instead, for when the input to the current macro is the result of another macro and thus will not have a source file?